tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/Utils/RulesConfiguration.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Sdk.Tools.GitHubEventProcessor.Constants;
+using System.IO;
+
+namespace Azure.Sdk.Tools.GitHubEventProcessor.Utils
+{
+    /// <summary>
+    /// Enum for whether or not a given rule is enabled (On) or disabled (Off).
+    /// JsonStringEnumConverter is necessary so the rules, when stored in the file, show On/Off instead
+    /// of 0/1 which makes things more readable.
+    /// </summary>
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public enum RuleState
+    {
+        On,
+        Off
+    }
+
+    /// <summary>
+    /// Class to hold the RulesConfiguration which includes methods to load the config, check whether
+    /// a given rule is enabled/disabled and report any missing rules.
+    /// </summary>
+    public class RulesConfiguration
+    {
+        public Dictionary<string, RuleState> Rules { get; set; }
+        public string RulesConfigFile { get; set; } = null;
+        public RulesConfiguration() 
+        {
+            Rules = new Dictionary<string, RuleState>();
+        }
+
+        /// <summary>
+        /// Overloaded constructor. If the configuration file location is passed in, load that otherwise
+        /// look in a well known location for the repository config file.
+        /// </summary>
+        /// <param name="configurationFile">Location of the configuration file or null, to look in the well known location for a repository.</param>
+        public RulesConfiguration(string configurationFile = null)
+        {
+            Rules = new Dictionary<string, RuleState>();
+            string configLoc = configurationFile;
+            if (configLoc == null)
+            {
+                // Load the config from the well known location, somewhere under the .github directory
+                // which is in the root of the repository
+                configLoc = DirectoryUtils.FindFileInRepository("event-processor.config", ".github");
```

Comment:
With so much effort spent on abstracting out constants to avoid magic strings, it seems odd that we'd use them here.

Reply:
Good call, these should be defined as a constant in the RulesConfiguration file. I'll update with the next commit.